src/sentry/api/bases/organization.py

```diff
@@ -242,30 +242,32 @@ class ControlSiloOrganizationEndpoint(Endpoint):
     def convert_args(
         self,
         request: Request,
-        organization_slug: str | int | None = None,
+        organization_id_or_slug: int | str | None = None,
         *args: Any,
         **kwargs: Any,
     ) -> tuple[tuple[Any, ...], dict[str, Any]]:
         if not subdomain_is_region(request):
             subdomain = getattr(request, "subdomain", None)
-            if subdomain is not None and subdomain != organization_slug:
+            if subdomain is not None and subdomain != organization_id_or_slug:
```

Comment:
Subdomain is never ID. Does this mean it would fail anytime we use the API with ID?

Reply:
I think this is related to my question here: https://sentry.slack.com/archives/C032E82D338/p1714605291155389, the git blame is related to customer domain stuff.

Reply:
tldr from the thread: this would fail if we get a request like `acme.sentry.io/organizations/12345/issues` because we wouldn't redirect the user to `http://12345.sentry.io/issues`. However, most of our customers will hit our endpoints with ids programmatically, so we should be good!

Reply:
https://github.com/getsentry/sentry/pull/70172